### ISSUE TITLE
SPH smoothing should be able to optionally switch to "gather" method

### ISSUE THREAD

**Issue by ngoldbaum** [yt-project/yt#1367]
Currently yt makes use of the Monaghan style "scatter" method to do SPH smoothing. Others (particularly users of Gadget) may prefer the "gather" method. It would be nice to make it so the core SPH smoothing operation could optionally switch to gather (possibly making use of the KDTree to do neighbor searching).

See the discussion in #1146

**Comment by chummels** [comment]
Agreed.  This enhancement will enable construction of an octree without empty cells, which can occur under certain circumstances using the "scatter" method.
